mousestyles/dynamics/tests/test_dynamics.py

```diff
@@ -98,49 +98,95 @@ def test_get_prob_matrix_small_interval():
+    time_list = [0, 1, 2]
+    with pytest.raises(ValueError) as excinfo:
+        mcmc_simulation(mat_list=time_list, n_per_int=10)
+    assert excinfo.value.args[0] == "items in mat_list should be numpy array"
+    # n_per_int is not integer
+    mat0 = np.zeros(16).reshape(4, 4)
+    np.fill_diagonal(mat0, val=1)
+    mat1 = np.zeros(16).reshape(4, 4)
+    mat1[0, 1] = 1
+    mat1[1, 0] = 1
+    mat1[2, 2] = 1
+    mat1[3, 3] = 1
+    mat_list_example = [mat0, mat1]
+    with pytest.raises(ValueError) as excinfo:
+        mcmc_simulation(mat_list=mat_list_example, n_per_int=0.5)
+    assert excinfo.value.args[0] == "n_per_int should be positive int"
+    # n_per_int negative integer
+    with pytest.raises(ValueError) as excinfo:
+        mcmc_simulation(mat_list=mat_list_example, n_per_int=-1)
+    assert excinfo.value.args[0] == "n_per_int should be positive int"
+
+
+def test_mcmc_simulation():
+    # Checking functions output the array
+    mat0 = np.zeros(16).reshape(4, 4)
+    np.fill_diagonal(mat0, val=1)
+    mat1 = np.zeros(16).reshape(4, 4)
+    mat1[0, 1] = 1
+    mat1[1, 0] = 1
+    mat1[2, 2] = 1
+    mat1[3, 3] = 1
+    mat_list_example = [mat0, mat1]
+    example = mcmc_simulation(mat_list_example, 10)
+    assert sum(example[:10]) == 0.
+    assert sum(example[10:]) == 5.
+    assert example[10] == 1.
+    assert example[11] == 0.
+
+
+def test_get_score_input():
+    # checking functions raise the correct errors for wrong input
+    # true_day is not numpy.array
+    with pytest.raises(ValueError) as excinfo:
+        get_score(true_day=0, simulated_day=np.zeros(13))
+    assert excinfo.value.args[0] == "true_day should be numpy array!"
+    # simulated_day is not numpy.array
+    with pytest.raises(ValueError) as excinfo:
+        get_score(true_day=np.zeros(13), simulated_day=0)
+    assert excinfo.value.args[0] == "simulated_day should be numpy array!"
+    # weight should be list
+    with pytest.raises(ValueError) as excinfo:
+        get_score(true_day=np.zeros(13), simulated_day=np.zeros(13),
+                  weight=0)
+    assert excinfo.value.args[0] == "weight should be list!"
+    # length of weight should be exactly 4
+    with pytest.raises(ValueError) as excinfo:
+        get_score(true_day=np.zeros(13), simulated_day=np.zeros(13),
+                  weight=[0])
+    assert excinfo.value.args[0] == "Length of weight should be 4!"
+    # check lengths of true_day and simulated_day
+    with pytest.raises(ValueError) as excinfo:
+        get_score(true_day=np.zeros(13), simulated_day=np.zeros(5))
+    error_message = "Length of simulated_day is smaller than true_day!"
+    assert excinfo.value.args[0] == error_message
+    # check all the weights are positive
+    with pytest.raises(ValueError) as excinfo:
+        get_score(true_day=np.zeros(13), simulated_day=np.zeros(13),
+                  weight=[-1, 2, 3, 4])
+    assert excinfo.value.args[0] == "All the weights should be positive!"

```

Comment:
will it be better to remove the ! at the end of each error message to stay consistent with other functions?